### PR TITLE
[gcp] docs/user: Add GCP installer user documentation

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

File: docs/user/gcp/dns.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# DNS Setup
+
+Google's Cloud DNS service is used by the OpenShift installer to configure cluster DNS resolution and provide name lookup
+for the cluster to the outside world. To use OpenShift, you must have created a public zone in Google Cloud DNS in
+the same project as your OpenShift cluster. You must also ensure the zone is "authoritative" for the domain. There are
+two ways to do this outlined below: root domain and subdomain. A root domain is `openshiftcorp.com`. A subdomain is of
```
Comment (abhinavdahiya) [inline docs/user/gcp/dns.md]: `example.com`

File: docs/user/gcp/dns.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+You can use [Google Domains](https://domains.google.com) to purchase a domain or use your preferred registrar.
+
+## Step 2: Create Public Zone
+
+Whether using a root domain or a subdomain, you must create a public DNS zone.
+
+[GCP: Creating a Public DNS Zone][create-dns-zone]
+
+To use the root domain, you'd create the DNS zone with the value `openshiftcorp.com`. To use a subdomain, you'd
```
Comment (abhinavdahiya) [inline docs/user/gcp/dns.md]: `example.com`

File: docs/user/gcp/iam.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+## Step 2: Assign Project Roles to your Service Account
+
+You need to assign the proper roles to the newly created service account so it can be used to create an OpenShift cluster.
+The minimum set of roles you can assign the service account are the following:
+- Compute Admin
+- DNS Administrator
+- Security Admin
+- Service Account Admin
+- Service Account User
```
Comment (abhinavdahiya) [inline docs/user/gcp/iam.md]: do we need service account admin and user? what about Logs Writer?

File: docs/user/gcp/iam.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Create a GCP service account using the console or the CLI
+
+[GCP: Creating a service account][sa-create]
+
+## Step 2: Assign Project Roles to your Service Account
+
+You need to assign the proper roles to the newly created service account so it can be used to create an OpenShift cluster.
+The minimum set of roles you can assign the service account are the following:
+- Compute Admin
```
Comment (abhinavdahiya) [inline docs/user/gcp/iam.md]: link to `https://cloud.google.com/iam/docs/understanding-roles#predefined_roles` maybe??

File: docs/user/gcp/install.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+### Running Cluster
+
+In your GCP project, there will be a new private DNS zone (for internal lookups)
+
+There will be six running VM instances in the Project.
+
+The nodes within the Virtual Network utilize the internal DNS and use the Router and Internal API load balancers. External/Internet
+access to the cluster use the Router and External API load balancers. Compute instances are spread equally across 3 availability
```
Comment (abhinavdahiya) [inline docs/user/gcp/install.md]: `across all RUNNING availability zones for the region`

**Comment by abhinavdahiya** [comment]
These look great!

File: docs/user/gcp/iam.md (hunk 3 above)
Comment (abhinavdahiya) [reply docs/user/gcp/iam.md]: Yeah, looking at https://cloud.google.com/iam/docs/understanding-roles#service-accounts-roles

it looks like we need `iam.serviceAccounts.actAs` from Service Account User, which is not in Service Account Admin :P

> I am less sure about the Logs Writer role, but will test it and see if we need it
+1

Comment (csrwng) [reply docs/user/gcp/iam.md]: Removed the Logs Writer role, confirmed that we don't need it

**Pull request merged**